### PR TITLE
Features/reminders flow tests

### PR DESCRIPTION
**Pull request by Sargatos**
- Created the tests for Reminders and Notifications.
- Added and modified Xrm Mocked Classes.
- Changed flow "Reminder - Daily - Create Daily Notifications" to only use CDS current environment.
- Changed flow "Reminder - Daily - Create Daily Notifications" to grab earlier reminder dates.
- Modified MultiLanguagePlugin to use RetrieveMultiple instead of Retrieve.

### CODE REVIEW

File: src/Compliance.Plugins/MultiLanguagePlugin.cs [hunk 1]
```diff
@@ -65,20 +66,26 @@ protected string UnpackName(LocalPluginContext localContext, string name)
...
+                    Criteria =
+                    {
+                        Conditions =
+                        {
+                            new ConditionExpression(userSettingsUserIdAttribute, ConditionOperator.Equal, localContext.PluginExecutionContext.InitiatingUserId)
+                        }
+                    }
+                };
+
+                EntityCollection userSettingsCollection = localContext.OrganizationService.RetrieveMultiple(queryExpression);
```
[the remark this answers is not in the document]
Comment (Sargatos) [reply src/Compliance.Plugins/MultiLanguagePlugin.cs]: Retrieve assumes that what you are looking for exist. If it doesn't it throws an exception. With RetrieveMultiple it returns an array that can be empty. We can then remove the try/catch that assigns the default value in its catch.

File: src/Compliance.WebResources/src/forms/ReminderMainForm.test.ts [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+            form.initializeComponents(mockContext);
+
+            // Act
+            mockContext.getFormContext().getAttribute("opc_notifyadditionalusers").fireOnChange();
+
+            // Assert
+            contextSpy.getFormContext().ui.tabs.get("tab_general").sections.get("section_additionalusers").getVisible().should.equal(false);
+        });
+    });
+});
```
[the remark this answers is not in the document]
Comment (Sargatos) [reply src/Compliance.WebResources/src/forms/ReminderMainForm.test.ts]: How should the opposite be included? ("it should hide the additional users section")
Another describe?

describe("when notify additional users is selected,"...
    it("it should display the additional users section"...
describe("when notify additional users is not selected,"...
    it("it should hide the additional users section"...

File: src/Compliance.WebResources/src/forms/ReminderMainForm.test.ts [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+            // Act
+            mockContext.getFormContext().data.entity.save();
+
+            // Assert
+            contextSpy.getFormContext().ui.getFormNotificationsLength().should.equal(1);
+            contextSpy.getFormContext().getControl("opc_notifycaseowner").getNotificationsLength().should.equal(1);
+        });
+
+        it("it should not display notifications if the user chooses to notify the case owner and has a case is related to the reminder", () => {
```
[the remark this answers is not in the document]
Comment (Sargatos) [reply src/Compliance.WebResources/src/forms/ReminderMainForm.test.ts]: Yes, the "has" is a mistake.
it should not display notifications if the user chooses to notify the case owner while a case is related to the reminder?

File: src/Compliance.WebResources/src/forms/ReminderMainForm.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (pierluc-ssoucy) [reply src/Compliance.WebResources/src/forms/ReminderMainForm.test.ts]: that seems to make sense yes. Just make sure that when tests are being executed, the test can be read in plain english.

**Pull request merged**